gettools.py

```diff
@@ -71,89 +72,154 @@ def convertpath(path):
-    os.remove(convertpath(dest + '/tools/com.vmware.fusion.tools.darwin.zip'))
-
-    # Download the darwinPre15.iso tgz file
-    print('Retrieving DarwinPre15 tools from: ' + urlpre15)
-    urlretrieve(urlpre15, convertpath(dest + '/tools/com.vmware.fusion.tools.darwinPre15.zip.tar'))
-
-    # Extract the tar to zip
-    tar = tarfile.open(convertpath(dest + '/tools/com.vmware.fusion.tools.darwinPre15.zip.tar'), 'r')
-    tar.extract('com.vmware.fusion.tools.darwinPre15.zip', path=convertpath(dest + '/tools/'))
-    tar.close()
-
-    # Extract the iso and sig files from zip
-    cdszip = zipfile.ZipFile(convertpath(dest + '/tools/com.vmware.fusion.tools.darwinPre15.zip'), 'r')
-    cdszip.extract('payload/darwinPre15.iso', path=convertpath(dest + '/tools/'))
-    cdszip.extract('payload/darwinPre15.iso.sig', path=convertpath(dest + '/tools/'))
-    cdszip.close()
-
-    # Move the iso and sig files to tools folder
-    shutil.move(convertpath(dest + '/tools/payload/darwinPre15.iso'),
-                convertpath(dest + '/tools/darwinPre15.iso'))
-    shutil.move(convertpath(dest + '/tools/payload/darwinPre15.iso.sig'),
-                convertpath(dest + '/tools/darwinPre15.iso.sig'))
-
-    # Cleanup working files and folders
-    shutil.rmtree(convertpath(dest + '/tools/payload'), True)
-    os.remove(convertpath(dest + '/tools/com.vmware.fusion.tools.darwinPre15.zip.tar'))
-    os.remove(convertpath(dest + '/tools/com.vmware.fusion.tools.darwinPre15.zip'))
-
+	# Check minimal Python version is 2.7
+	if sys.version_info < (2, 7):
+		sys.stderr.write('You need Python 2.7 or later\n')
+		sys.exit(1)
+
+	dest = os.path.dirname(os.path.abspath(__file__))
+
+	# Re-create the tools folder
+	shutil.rmtree(dest + '/tools', True)
+	os.mkdir(dest + '/tools')
+
+	parser = CDSParser()
+
+	# Last published version doesn't ship with darwin tools
+	# so in case of error get it from the core.vmware.fusion.tar
+	print('Trying to get tools from the packages folder...')
+
+	# Setup url and file paths
+	url = 'http://softwareupdate.vmware.com/cds/vmw-desktop/fusion/'
+
+	# Get the list of Fusion releases
+	# And get the last item in the ul/li tags
+	
+	response = urlopen(url)
+	html = response.read()
+	parser.clean()
+	parser.feed(str(html))
+	url = url + parser.HTMLDATA[-1] + '/'
+	parser.clean()
+
+	# Open the latest release page
+	# And build file URL
+	response = urlopen(url)
+	html = response.read()
+	parser.feed(str(html))
+	
+	lastVersion = parser.HTMLDATA[-1]
+	
+	urlpost15 = url + lastVersion + '/packages/com.vmware.fusion.tools.darwin.zip.tar'
+	urlpre15 = url + lastVersion + '/packages/com.vmware.fusion.tools.darwinPre15.zip.tar'
+	parser.clean()
+
+	# Download the darwin.iso tgz file
+	print('Retrieving Darwin tools from: ' + urlpost15)
+	try:
+		# Try to get tools from packages folder
+		urlretrieve(urlpost15, convertpath(dest + '/tools/com.vmware.fusion.tools.darwin.zip.tar'))
```

Comment:
Seems that it doesn't throw if directory not exist. 😞

Reply:
Well, I last used it two months ago,
I did another test just now, the previous url can be accessed.
And the exe file by pyinstaller cannot throw 404 error.